src/librustc_typeck/astconv/generics.rs

```diff
@@ -0,0 +1,601 @@
+//! The `AstConvGeneric` trait is a subtrait of the `AstConv` trait. The methods present
+//! here help when handling generic types from the AST reprsentation.
+
+use crate::astconv::{
+    AstConv, ExplicitLateBound, GenericArgCountMismatch, GenericArgCountResult, GenericArgPosition,
+};
+use rustc_ast::ast::ParamKindOrd;
+use rustc_errors::{pluralize, struct_span_err, DiagnosticId, ErrorReported};
+use rustc_hir as hir;
+use rustc_hir::def_id::DefId;
+use rustc_hir::{GenericArg, GenericArgs};
+use rustc_middle::ty::{
+    self, subst, subst::SubstsRef, GenericParamDef, GenericParamDefKind, Ty, TyCtxt,
+};
+use rustc_session::{lint::builtin::LATE_BOUND_LIFETIME_ARGUMENTS, Session};
+use rustc_span::{symbol::kw, MultiSpan, Span};
+
+use smallvec::SmallVec;
+
+pub trait AstConvGeneric<'tcx>: AstConv<'tcx> {}
+
+impl<'tcx> dyn AstConvGeneric<'tcx> {
```

Comment:
Instead of creating a second trait, you can also use
```suggestion
impl<'tcx> dyn AstConv<'tcx> {
```

here again, as you can have multiple impl blocks for the same thing, even in different modules

Reply:
Oh thanks, I didn't know that. Fixing it.